src/ILCompiler.Compiler/src/Compiler/MultiFileCompilationModuleGroup.cs

```diff
@@ -77,7 +79,16 @@ private void AddCompilationRootsForMultifileLibrary(EcmaModule module, IRootingS
                 if (type.IsDelegate || type.ContainsGenericVariables)
                     continue;
 
-                rootProvider.AddCompilationRoot(type, "Library module type");
+                try
+                {
+                    rootProvider.AddCompilationRoot(type, "Library module type");
+                }
+                catch (TypeSystemException)
+                {
+                    // Swallow type load exceptions while rooting
+                    continue;
```

Comment:
Can you add the same TODO comments we have around this catch block as we have in `RyuJitCompilation`? (We'll probably want (suppressible) warnings, plus a strict mode where missing references are fatal.)